Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.py

```diff
@@ -193,35 +195,419 @@ def delete_indicator_command():
+
+    Returns:
+        bool. Weather the incident has passed the filter or not
+    """
+    if len(abuse_disposition_values) == 0:
+        return True
+
+    for incident_field in incident['incident_field_values']:
+        if incident_field['name'] == "Abuse Disposition":
+            if incident_field['value'] in abuse_disposition_values:
+                return True
+
+    return False
+
+
+def filter_incidents(incidents_list):
+    """Filters the incidents list by two existing filters.
+
+    Returns:
+        list. The filtered incidents list
+    """
+    filtered_incidents_list = []
+    params = demisto.params()
+    sources_list = argToList(params.get('event_sources'))
+    abuse_disposition_values = argToList(params.get('abuse_disposition'))
+
+    if not sources_list and not abuse_disposition_values:
+        return incidents_list
+
+    for incident in incidents_list:
+        if pass_sources_list_filter(incident, sources_list):
+            if pass_abuse_disposition_filter(incident, abuse_disposition_values):
+                filtered_incidents_list.append(incident)
+
+    return filtered_incidents_list
+
+
+def get_incidents_request(params):
+    """Perform an API request to get incidents from ProofPoint.
+    Args:
+        params(dict): The params of the request
+    Returns:
+        list. The incidents returned from the API call
+    """
+    fullurl = BASE_URL + 'api/incidents'
+    incidents_list = requests.get(
+        fullurl,
+        headers={
+            'Content-Type': 'application/json',
+            'Authorization': API_KEY
+        },
+        params=params,
+        verify=VERIFY_CERTIFICATE
+    )
+
+    if incidents_list.status_code < 200 or incidents_list.status_code >= 300:
+        if incidents_list.status_code == 502 or incidents_list.status_code == 504:
+            return_error('The operation failed. There is a possibility you are trying to get too many incidents.\n'
+                         'You may consider adding a filter argument to the command.\n'
+                         'URL: {}, StatusCode: {}, Response: {}'.format(fullurl, incidents_list.status_code,
+                                                                        incidents_list.text))
+        else:
+            return_error('The operation failed. URL: {}, StatusCode: {}, '
+                         'Response: {}'.format(fullurl, incidents_list.status_code, incidents_list.text))
+    return incidents_list.json()
+
+
+def fetch_incidents_command():
+    """
+        Fetches incidents from the ProofPoint API.
+    """
+    integration_params = demisto.params()
+    last_fetch = demisto.getLastRun().get('last_fetch', {})
+    incidents_states = integration_params.get('states')
+    for state in incidents_states:
+        if not last_fetch.get(state):
+            last_fetch[state] = FETCH_TIME
+
+    incidents = []
+
+    for state in incidents_states:
+        request_params = {
+            'created_after': last_fetch[state],
```

Comment:
`last_fetch` can be an empty dict, so it might fail on KeyError

Reply:
No it can't, because of the for loop in line 435